core/services/relay/evm/functions.go

```diff
@@ -36,35 +43,76 @@ func (p *functionsProvider) ContractTransmitter() types.ContractTransmitter {
 	return p.contractTransmitter
 }
 
-func NewFunctionsProvider(chainSet evm.ChainSet, rargs relaytypes.RelayArgs, pargs relaytypes.PluginArgs, lggr logger.Logger, ethKeystore keystore.Eth, pluginType functionsRelay.FunctionsPluginType) (relaytypes.Plugin, error) {
-	configWatcher, err := newFunctionsConfigProvider(pluginType, chainSet, rargs, lggr)
+func (p *functionsProvider) LogPollerWrapper() evmRelayTypes.LogPollerWrapper {
+	return nil
+}
+
+func (p *functionsProvider) Start(ctx context.Context) error {
+	return p.StartOnce("FunctionsProvider", func() error {
+		if err := p.configWatcher.Start(ctx); err != nil {
+			return err
+		}
+		return p.logPollerWrapper.Start(ctx)
+	})
+}
+
+func (p *functionsProvider) Close() error {
+	return p.StopOnce("FunctionsProvider", func() (err error) {
+		err = multierr.Combine(err, p.logPollerWrapper.Close())
+		err = multierr.Combine(err, p.configWatcher.Close())
+		return
+	})
+}
+
+// Forward all calls to the underlying configWatcher
```

Comment:
IIRC, eventually we can support multiple configWatchers and our LogPollerWrapper can forward calls to the whichever one we need.